server/src/main/scala/org/http4s/server/middleware/RequestId.scala

```diff
@@ -100,6 +99,6 @@ object RequestId {
         headerName: CIString = requestIdHeader,
         genReqId: F[UUID]
     )(httpRoutes: HttpRoutes[F]): HttpRoutes[F] =
-      RequestId.apply(OptionT.liftK[F], headerName, genReqId)(httpRoutes)
+      RequestId.apply(OptionT.liftK.andThen(Kleisli.liftK[OptionT[F, *], Request[F]]), headerName, genReqId)(httpRoutes)
```

Comment:
Inference is still garbage without this one.  The ones without the `~>` seem to do alright.

We wouldn't need it if people would just bring a `G[UUID]`.  With a bespoke `HttpRoutes` with a `Sync` instance, that's easy.  With an explicit stack, it looks like, uh, this?